seed/static/seed/js/controllers/inventory_map_controller.js

```diff
@@ -0,0 +1,409 @@
+        var opacity = Math.max(0.20,total_eui/hexagon_size);
+
+        return hexagonStyle(opacity);
+      };
+
+      $scope.hexbin_layer = new ol.layer.Vector({
+        source: hexbinSource(),
+        zIndex: 1,  // Note: This is used for layer toggling.
+        opacity: 0.8,
+        style:  hexbinStyle,
+      })
+
+      // Render map
+      $scope.map = new ol.Map({
+        target: 'map',
+        layers: [base_layer, $scope.hexbin_layer, $scope.points_layer]
+      });
+
+      // Toggle layers
+
+      // If a layer's z-index is changed, it should be changed here as well.
+      var layer_at_z_index = {
+        0: base_layer,
+        1: $scope.hexbin_layer,
+        2: $scope.points_layer,
+      }
+
+      $scope.layerVisible = function(z_index) {
+        var layers = $scope.map.getLayers().array_
+        var z_indexes = _.map(layers, function(layer) {
+          return layer.values_.zIndex;
+        });
+
+        return z_indexes.includes(z_index);
+      };
+
+      $scope.toggle_layer = function(z_index) {
+        if ($scope.layerVisible(z_index)) {
+          $scope.map.removeLayer(layer_at_z_index[z_index]);
+        } else {
+          $scope.map.addLayer(layer_at_z_index[z_index]);
+        }
+      };
+
+      // Popup
+      var popup_element = document.getElementById('popup-element');
+
+      // Define overlay attaching html element
+      var popup_overlay = new ol.Overlay({
+        element: popup_element,
+        positioning: 'bottom-center',
+        stopEvent: false,
+        autoPan: true,
+        autoPanMargin: 75,
+        offset: [0, -135]
+      });
+      $scope.map.addOverlay(popup_overlay);
+
+      var detailPageIcon = function(point_info) {
+        var link_html = ''
+        var icon_html = '<i class="ui-grid-icon-info-circled"></i>'
+
+        if ($scope.inventory_type == 'properties') {
+          link_html = '<a href="#/properties/' +
+            point_info.property_view_id +
+            '">' +
+            icon_html +
+            '</a>'
+        } else {
+          link_html = '<a href="#/taxlots/' +
+            point_info.property_view_id +
+            '">' +
+            icon_html +
+            '</a>'
+        }
+
+        return link_html
+      };
+
+      var showPointInfo = function (point) {
+        var pop_info = point.getProperties();
+        var address_line_1_key = _.find(_.keys(pop_info), function(key) {
+          return _.startsWith(key, "address_line_1")
+        });
+
+        var coordinates = point.getGeometry().getCoordinates();
+
+        popup_overlay.setPosition(coordinates);
+        $(popup_element).popover({
+          placement: 'top',
+          html: true,
+          selector: true,
+          content: pop_info[address_line_1_key] + detailPageIcon(pop_info)
+        });
+
+        $(popup_element).popover('show');
+      };
+
+      // Define point/cluster click event
+      // TODO Display info doesn't change bug when popover already shown.
```

Comment:
Cleanup item - This is a bug I couldn't quite figure out. I plan to revisit this again to try to resolve.